doc/source/overview/config_files.md

```diff
@@ -0,0 +1,752 @@
+objects defined at runtime can be registered using `alibi_detect.utils.registry`, allowing them to be specified 
+in the config file without having to serialise them. For convenience a number of Alibi Detect functions such as 
+[preprocess_drift](../api/alibi_detect.cd.tensorflow.preprocess.rst) are also pre-registered. 
+
+- **Dictionaries**: More complex artefacts are specified via nested dictionaries with a `src` field and 
+additional option/setting fields, which are sometimes further nested dictionaries themselves. See 
+[Artefact dictionaries](dictionaries) for further details.
+
+The following table shows the allowable formats for all artefacts that can be specified in a config file.
+
+```{table} Possible artefact formats
+:name: all-artefacts-table
+
+|Field                     |.npy file  |.dill file  |Registry|Dictionary| 
+|:-------------------------|:---------:|:----------:|:------:|:--------:|
+|`x_ref`                   |✔          |            |        |          |
+|`optimizer`               |           |✔           |✔       |          |
+|`reg_loss_fn`             |           |✔           |✔       |          |
+|`model`                   |           |            |✔       |✔         |
+|`preprocess_fn`           |           |✔           |✔       |✔         |
+|`preprocess_fn.model`     |           |            |✔       |✔         |
+|`preprocess_fn.embedding` |           |            |✔       |✔         |
+|`preprocess_fn.tokenizer` |           |            |✔       |✔         |
+|`dataset`                 |           |✔           |✔       |✔         |
+|`kernel`                  |           |✔           |✔       |✔         |
+|`kernel.proj`             |           |            |✔       |✔         |
+|`kernel.kernel_a`         |           |✔           |✔       |✔         |
+|`kernel.kernel_b`         |           |✔           |✔       |✔         |
+|`initial_diffs`           |✔          |            |        |          |
+```
+
+```{note}
+When TensorFlow, PyTorch, or scikit-learn `model`'s are specified, the model type should be specified via the `backend` field in the *config.toml*.
+For example, if `model` is a TensorFlow model, set `backend='tensorflow'. This is the case even for detectors that don't take a `backend` kwarg.
+```
+
+(dictionaries)=
+### Artefact dictionaries
+
+The period in the `preprocess_fn.model` field in the {ref}`all-artefacts-table` table indicates that `model` is in 
+fact a nested field. In other words, `preprocess_fn` is an [Artefact dictionary](dictionaries), and `model` is a field 
+within the `preprocess_fn` dictionary. This layout is required to specify more complex artefacts such as 
+`preprocess_fn`, which themselves rely on other artefacts. TOML files are highly flexible when it comes to specifying 
+dictionaries; each section in a TOML file, demarcated by section headers enclosed in [ ] brackets, is actually a Python 
```

Comment:
May be better to link to TOML docs that explain how to define sections instead of explaining it bellow?